x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/googlevertexai/completion/GoogleVertexAiChatCompletionTaskSettings.java

```diff
@@ -0,0 +1,129 @@
+    public static final String NAME = "google_vertex_ai_chatcompletion_task_settings";
+
+    private final ThinkingConfig thinkingConfig;
+
+    public static final GoogleVertexAiChatCompletionTaskSettings EMPTY_SETTINGS = new GoogleVertexAiChatCompletionTaskSettings();
+    private static final ThinkingConfig EMPTY_THINKING_CONFIG = new ThinkingConfig();
+
+    public GoogleVertexAiChatCompletionTaskSettings() {
+        thinkingConfig = EMPTY_THINKING_CONFIG;
+    }
+
+    public GoogleVertexAiChatCompletionTaskSettings(ThinkingConfig thinkingConfig) {
+        this.thinkingConfig = Objects.requireNonNullElse(thinkingConfig, EMPTY_THINKING_CONFIG);
+    }
+
+    public GoogleVertexAiChatCompletionTaskSettings(StreamInput in) throws IOException {
+        if (in.getTransportVersion().onOrAfter(TransportVersions.GEMINI_THINKING_BUDGET_ADDED)) {
+            thinkingConfig = new ThinkingConfig(in);
+        } else {
+            thinkingConfig = EMPTY_THINKING_CONFIG;
+        }
+    }
+
+    public static GoogleVertexAiChatCompletionTaskSettings fromMap(Map<String, Object> taskSettings) {
+        ValidationException validationException = new ValidationException();
+
+        // Extract optional thinkingConfig settings
+        ThinkingConfig thinkingConfig = ThinkingConfig.fromMap(taskSettings, validationException);
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+
+        return new GoogleVertexAiChatCompletionTaskSettings(thinkingConfig);
+    }
+
+    public static GoogleVertexAiChatCompletionTaskSettings of(
+        GoogleVertexAiChatCompletionTaskSettings originalTaskSettings,
+        GoogleVertexAiChatCompletionTaskSettings newTaskSettings
+    ) {
+        ThinkingConfig thinkingConfig = newTaskSettings.thinkingConfig().isEmpty()
+            ? originalTaskSettings.thinkingConfig()
+            : newTaskSettings.thinkingConfig();
+        return new GoogleVertexAiChatCompletionTaskSettings(thinkingConfig);
+    }
+
+    public ThinkingConfig thinkingConfig() {
+        return thinkingConfig;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return thinkingConfig.isEmpty();
+    }
+
+    @Override
+    public TaskSettings updatedTaskSettings(Map<String, Object> newSettings) {
+        GoogleVertexAiChatCompletionTaskSettings newTaskSettings = GoogleVertexAiChatCompletionTaskSettings.fromMap(
+            new HashMap<>(newSettings)
+        );
+        return GoogleVertexAiChatCompletionTaskSettings.of(this, newTaskSettings);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return TransportVersions.GEMINI_THINKING_BUDGET_ADDED;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (out.getTransportVersion().onOrAfter(TransportVersions.GEMINI_THINKING_BUDGET_ADDED)) {
```

Comment:
Since we're introducing this class we don't need to check the version here. If we wanted to protect against a mixed cluster scenario we'd need to do this check when we read the named writeable but we haven't been handling that scenario because it's short lived.